packages/ingest/src/FtpProviderClient.ts

```diff
@@ -99,37 +132,38 @@ class FtpProviderClient {
   /**
    * List all files from a given endpoint
    * @param {string} path - path to list
-   * @param {number} _counter - recursive attempt counter
+   * @param {number} counter - recursive attempt counter
    * @returns {Promise} promise of contents
    * @private
    */
-  async _list(path, _counter = 0) {
-    let counter = _counter;
+  async _list(path: string, counter = 0): Promise<FtpProviderClientListItem[]> {
     const client = await this.buildFtpClient();
-    return new Promise((resolve, reject) => {
+    return new Promise<FtpProviderClientListItem[]>((resolve, reject) => {
       client.on('error', this.errorHandler.bind(this, reject));
-      client.ls(path, (err, data) => {
+      client.ls(path, (err: Error | ListError, data) => {
         if (err) {
-          const message = err.message || err.text;
+          const message = isJSFtpError(err) ? err.text : err.message;
           if (message && message.includes('Timed out') && counter < 3) {
             log.error(`Connection timed out while listing ${path}. Retrying...`);
-            counter += 1;
-            return this._list(path, counter).then((r) => {
-              log.info(`${counter} retry succeeded`);
+            return this._list(path, counter + 1).then((r) => {
```

Comment:
Does this really behave the same without the `counter += 1`? I guess we don't need to update the value of `counter` in this scope, just pass the incremented value into the recursion?

Reply:
Correct.